Add support to automatically generate ARM templates and Deploy to Azu…

KevinRabun opened the pull request:
…re buttons

ugreg commented:
First off this is a fantastic contribution, thanks @KevinRabun !

Can we also add this to the action as well? This way we can fun the workflow on any branch we want manually?

```json
on:
  workflow_dispatch
```

KevinRabun commented:
> First off this is a fantastic contribution, thanks @KevinRabun !
> 
> Can we also add this to the action as well? This way we can fun the workflow on any branch we want manually?
> 
> ```json
> on:
>   workflow_dispatch
> ```

Great callout! Added and updated

ugreg commented:
@KevinRabun do these Excel documents need to be modified in order to generate this URL?

![image](https://github.com/user-attachments/assets/c6652d59-1c30-4f17-bc45-58832075d246)
